Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -3009,4 +3006,173 @@ void SoftwareGLContext::sync_device_texcoord_config()
     m_rasterizer.set_options(options);
 }
 
+void SoftwareGLContext::gl_lightf(GLenum light, GLenum pname, GLfloat param)
+{
+    RETURN_WITH_ERROR_IF(m_in_draw_state, GL_INVALID_OPERATION);
+    RETURN_WITH_ERROR_IF(light > GL_LIGHT7, GL_INVALID_ENUM);
+    RETURN_WITH_ERROR_IF(!(pname == GL_CONSTANT_ATTENUATION || pname == GL_LINEAR_ATTENUATION || pname == GL_QUADRATIC_ATTENUATION || pname != GL_SPOT_EXPONENT || pname != GL_SPOT_CUTOFF), GL_INVALID_ENUM);
+
+    auto& device_light = m_rasterizer.light(light - GL_LIGHT0);
+    switch (pname) {
+    case GL_CONSTANT_ATTENUATION:
+        device_light.constant_attenuation = param;
+        break;
+    case GL_LINEAR_ATTENUATION:
+        device_light.linear_attenuation = param;
+        break;
+    case GL_QUADRATIC_ATTENUATION:
+        device_light.quadratic_attenuation = param;
+        break;
+    case GL_SPOT_EXPONENT:
+        device_light.spotlight_exponent = param;
+        break;
+    case GL_SPOT_CUTOFF:
+        device_light.spotlight_cutoff_angle = param;
+        break;
+    default:
+        VERIFY_NOT_REACHED();
+    }
+}
+
+void SoftwareGLContext::gl_lightfv(GLenum light, GLenum pname, const GLfloat* params)
```

Comment:
This should also work inside of a list

Reply:
I believe I've already added this to the right place for it to be executed in a list??

https://github.com/SerenityOS/serenity/pull/11764/files/0b92a7a66a769f8f875d114d589a51b5f22c6978#diff-0dbbc803292909480f93f2043b97891875af950652570513caa853651184b78aR370